tests/test_config.py

```diff
@@ -369,6 +369,28 @@ def generate_subcommand_test_cases(
         exp_config_type=RunConfig,
         exp_config_values={"fallback_url": "foobar.com"},
     ),
+    # health-endpoint
+    ConfigTestCase(
+        case="Run: health-endpoint unspecified",
+        args=["run"],
+        legacy_args=[],
+        exp_config_type=RunConfig,
+        exp_config_values={"health_endpoint": DEFAULTS.HEALTH_ENDPOINT},
+    ),
+    ConfigTestCase(
+        case="Run: health-endpoint specified",
+        args=["run", "--health-endpoint", "/healthz"],
+        legacy_args=["--health-endpoint", "/healthz"],
+        exp_config_type=RunConfig,
+        exp_config_values={"health_endpoint": "/healthz"},
+    ),
+    ConfigTestCase(
+        case="Run: health-endpoint specified but invalid",
+        args=["run", "--health-endpoint", "/health!"],
+        legacy_args=["--health-endpoint", "/health!"],
+        exp_config_type=RunConfig,
+        exp_config_values={"health_endpoint": DEFAULTS.HEALTH_ENDPOINT},
```

Comment:
This case should raise an error, similar to the `case=f"Invalid hash algo: {val}"`, as this would be a more straightforward outcome imho :) Linked to a comment above on `health_endpoint_arg`. And for inspiration on how to set the error cases up, see `_CONFIG_ERROR_CASES`.

Reply:
[raise SystemExit when using invalid args](https://github.com/pypiserver/pypiserver/pull/442/commits/edcf5778e732440f506b1ec1222fe7d78d0fc1c1)